### ISSUE TITLE
[Bug]: load diskann io_getevents() failed

### ISSUE THREAD

**Issue by jaime0815** [milvus-io/milvus#20869]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

```
2022-11-28T03:35:39.015704915+08:00 stdout F 2022-11-27 19:35:39,015 | INFO | default | [SEGCORE][N6milvus7storage17MinioChunkManagerE::MinioChunkManager][milvus] init MinioChunkManager with parameter[endpoint: 'mdp-203-n-minio:9000', access_key:'minioadmin', access_value:'minioadmin', default_bucket_name:'milvus-bucket', use_secure:'false']
2022-11-28T03:35:39.019622205+08:00 stdout F 2022-11-27 19:35:39,019 | INFO | default | [SEGCORE][N6milvus10ThreadPoolE::ThreadPool][milvus] Thread pool's worker num:480
2022-11-28T03:35:39.129033549+08:00 stdout F 2022-11-27 19:35:39,128 | INFO | default | [KNOWHERE][Prepare][milvus] Loading PQ from disk.
2022-11-28T03:35:39.129064915+08:00 stdout F 2022-11-27 19:35:39,128 | INFO | default | L2: Using AVX2 distance computation DistanceL2Float
2022-11-28T03:35:39.129069913+08:00 stdout F 2022-11-27 19:35:39,128 | INFO | default | L2: Using AVX2 distance computation DistanceL2Float
2022-11-28T03:35:39.129353439+08:00 stdout F 2022-11-27 19:35:39,129 | INFO | default | PQ Pivots: #ctrs: 256, #dims: 128, #chunks: 63
2022-11-28T03:35:39.12953857+08:00 stdout F 2022-11-27 19:35:39,129 | INFO | default | Loaded PQ centroids and in-memory compressed vectors. #points: 1497 #dim: 128 #aligned_dim: 128 #chunks: 63
2022-11-28T03:35:39.129573758+08:00 stdout F 2022-11-27 19:35:39,129 | INFO | default | Disk-Index File Meta-data: # nodes per sector: 5, max node len (bytes): 740, max node degree: 56
2022-11-28T03:35:39.129580627+08:00 stdout F 2022-11-27 19:35:39,129 | INFO | default | Setting up thread-specific contexts for nthreads: 64
2022-11-28T03:35:39.142731269+08:00 stdout F 2022-11-27 19:35:39,142 | INFO | default | Loading centroid data from medoids vector data of 1 medoid(s)
2022-11-28T03:35:39.145294951+08:00 stdout F 2022-11-27 19:35:39,145 | INFO | default | [KNOWHERE][Prepare][milvus] Caching 85 sample nodes around medoid(s).
2022-11-28T03:35:39.178730025+08:00 stderr F terminate called after throwing an instance of 'diskann::ANNException'
2022-11-28T03:35:39.178756399+08:00 stderr F   what():  [FUNC: void {anonymous}::execute_io(io_context_t, uint64_t, int, const std::vector<AlignedRead>&, uint64_t)][FILE: /go/src/github.com/milvus-io/milvus/cmake_build/3rdparty_download/knowhere-subbuild/src/knowhere_ep/thirdparty/DiskANN/src/linux_aligned_file_reader.cpp][LINE: 76]io_getevents() failed; returned 5, expected=6, ernno=0=Unknown error -5, try #1
```

Latest: https://jenkins.milvus.io:18080/blue/organizations/jenkins/Milvus%20Nightly%20CI/detail/master/203/pipeline/187
milvus: https://github.com/milvus-io/milvus/commit/f588a73719378f532734143b72fcca383eed2eca
pymilvus: 2.3.0.dev15
log: [artifacts-milvus-distributed-pulsar-nightly-203-pymilvus-e2e-logs.tar.gz](https://jenkins.milvus.io:18080/job/Milvus%20Nightly%20CI/job/master/203/artifact/artifacts-milvus-distributed-pulsar-nightly-203-pymilvus-e2e-logs.tar.gz)

**Comment by xige-16** [comment]
@liliu-z  please help

**Comment by liliu-z** [comment]
same as https://github.com/milvus-io/milvus/issues/20803 @hhy3

**Comment by jaime0815** [comment]
```
2022-11-29T05:10:46.515682268+08:00 stderr F terminate called after throwing an instance of 'diskann::ANNException'
2022-11-29T05:10:46.51570737+08:00 stderr F   what():  [FUNC: void {anonymous}::execute_io(io_context_t, uint64_t, int, const std::vector<AlignedRead>&, uint64_t)][FILE: /go/src/github.com/milvus-io/milvus/cmake_build/3rdparty_download/knowhere-subbuild/src/knowhere_ep/thirdparty/D
iskANN/src/linux_aligned_file_reader.cpp][LINE: 76]io_getevents() failed; returned 4, expected=6, ernno=2=Unknown error -4, try #1
2022-11-29T05:10:46.51779531+08:00 stderr F SIGABRT: abort
2022-11-29T05:10:46.5178107+08:00 stderr F PC=0x7f3ea26c900b m=87 sigcode=18446744073709551610
2022-11-29T05:10:46.517816011+08:00 stderr F signal arrived during cgo execution
2022-11-29T05:10:46.517819631+08:00 stderr F
2022-11-29T05:10:46.517823835+08:00 stderr F goroutine 76732 [syscall]:
2022-11-29T05:10:46.517836595+08:00 stderr F runtime.cgocall(0x2e82060, 0xc007092c38)
2022-11-29T05:10:46.517842151+08:00 stderr F    /usr/local/go/src/runtime/cgocall.go:157 +0x5c fp=0xc007092c10 sp=0xc007092bd8 pc=0x135837c
2022-11-29T05:10:46.517846815+08:00 stderr F github.com/milvus-io/milvus/internal/querynode._Cfunc_AppendIndex(0x7f3c9dc21300, 0x7f3cc32032f0)
2022-11-29T05:10:46.517854665+08:00 stderr F    _cgo_gotypes.go:172 +0x55 fp=0xc007092c38 sp=0xc007092c10 pc=0x2ce41f5
2022-11-29T05:10:46.517859095+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*LoadIndexInfo).appendIndexData.func5(0x7f3c9f00d260?, 0x3ce06a6?)
2022-11-29T05:10:46.51786331+08:00 stderr F     /go/src/github.com/milvus-io/milvus/internal/querynode/load_index_info.go:198 +0x65 fp=0xc007092c78 sp=0xc007092c38 pc=0x2cedfc5
2022-11-29T05:10:46.5178707+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*LoadIndexInfo).appendIndexData(0xc007092e00?, {0x0, 0x0, 0xc00663ae50?}, {0xc006476500, 0x7, 0x7f3ca480c130?})
2022-11-29T05:10:46.517884165+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/querynode/load_index_info.go:198 +0x145 fp=0xc007092d20 sp=0xc007092c78 pc=0x2cedce5
2022-11-29T05:10:46.517914568+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*LoadIndexInfo).appendLoadIndexInfo(0x7c?, {0x0, 0x0, 0x0}, 0xc006e1c140, 0x0?, 0xa?, 0x203000?, 0x203000?)
2022-11-29T05:10:46.517926128+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/querynode/load_index_info.go:122 +0x3b5 fp=0xc007092e70 sp=0xc007092d20 pc=0x2ced0d5
2022-11-29T05:10:46.517963696+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*Segment).segmentLoadIndexData(0xc006da16c0, {0x0, 0x0, 0x0}, 0xc006e1c140, 0x0?)
2022-11-29T05:10:46.517995011+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/querynode/segment.go:957 +0xf3 fp=0xc0070930b0 sp=0xc007092e70 pc=0x2d04eb3
2022-11-29T05:10:46.518017219+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).loadFieldIndexData(0xc000444180, {0x40f05b8?, 0xc006e187b0}, 0xc006da16c0, 0xc006e1c140)
2022-11-29T05:10:46.51802263+08:00 stderr F     /go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:518 +0x5e5 fp=0xc007093630 sp=0xc0070930b0 pc=0x2cb2365
2022-11-29T05:10:46.518047036+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).loadIndexedFieldData(0x394f080?, {0x40f05b8, 0xc006e187b0}, 0xc006da16c0, 0x7f3cd03eefff?)
2022-11-29T05:10:46.518068458+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:457 +0x11b fp=0xc0070937a8 sp=0xc007093630 pc=0x2cb1afb
2022-11-29T05:10:46.518096147+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).loadFiles(0xc000444180, {0x40f05b8, 0xc006e187b0}, 0xc006da16c0, 0xc00195ab60)
2022-11-29T05:10:46.518119159+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:274 +0xc7e fp=0xc007093be0 sp=0xc0070937a8 pc=0x2cafafe
2022-11-29T05:10:46.518134837+08:00 stderr F github.com/milvus-io/milvus/internal/querynode.(*segmentLoader).LoadSegment.func3(0x0?)
2022-11-29T05:10:46.518158289+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/querynode/segment_loader.go:176 +0xe5 fp=0xc007093dc0 sp=0xc007093be0 pc=0x2cae8c5
2022-11-29T05:10:46.51816383+08:00 stderr F github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel.func3(0x0)
2022-11-29T05:10:46.518167443+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:83 +0x16d fp=0xc007093fc8 sp=0xc007093dc0 pc=0x1efe7cd
2022-11-29T05:10:46.518171308+08:00 stderr F github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel.func4()
2022-11-29T05:10:46.518175007+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:101 +0x2a fp=0xc007093fe0 sp=0xc007093fc8 pc=0x1efe62a
2022-11-29T05:10:46.518179794+08:00 stderr F runtime.goexit()
2022-11-29T05:10:46.518191017+08:00 stderr F    /usr/local/go/src/runtime/asm_amd64.s:1571 +0x1 fp=0xc007093fe8 sp=0xc007093fe0 pc=0x13c1661
2022-11-29T05:10:46.518195122+08:00 stderr F created by github.com/milvus-io/milvus/internal/util/funcutil.ProcessFuncParallel
2022-11-29T05:10:46.518204018+08:00 stderr F    /go/src/github.com/milvus-io/milvus/internal/util/funcutil/parallel.go:70 +0x1aa
2022-11-29T05:10:46.518209769+08:00 stderr F
2022-11-29T05:10:46.518216011+08:00 stderr F goroutine 1 [chan receive, 71 minutes]:
2022-11-29T05:10:46.518231502+08:00 stderr F github.com/milvus-io/milvus/cmd/roles.(*MilvusRoles).Run(0xc000559e50, 0x0, {0x0, 0x0})
2022-11-29T05:10:46.518249682+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/roles/roles.go:306 +0xb32
2022-11-29T05:10:46.518256043+08:00 stderr F github.com/milvus-io/milvus/cmd/milvus.(*run).execute(0xc0008c02a0, {0xc0001a8060?, 0x3, 0x3}, 0xc00004c240)
2022-11-29T05:10:46.518260459+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/milvus/run.go:120 +0x733
2022-11-29T05:10:46.518264741+08:00 stderr F github.com/milvus-io/milvus/cmd/milvus.RunMilvus({0xc0001a8060?, 0x3, 0x3})
2022-11-29T05:10:46.518280777+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/milvus/milvus.go:60 +0x21e
2022-11-29T05:10:46.518397065+08:00 stderr F main.main()
2022-11-29T05:10:46.518416302+08:00 stderr F    /go/src/github.com/milvus-io/milvus/cmd/main.go:26 +0x2e
2022-11-29T05:10:46.518449043+08:00 stderr F
2022-11-29T05:10:46.518456993+08:00 stderr F goroutine 102 [chan receive]:
2022-11-29T05:10:46.518462322+08:00 stderr F github.com/panjf2000/ants/v2.(*Pool).purgePeriodically(0xc0001d7500)
2022-11-29T05:10:46.518466514+08:00 stderr F    /go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.8/pool.go:69 +0x8b
2022-11-29T05:10:46.518525765+08:00 stderr F created by github.com/panjf2000/ants/v2.NewPool
2022-11-29T05:10:46.518531355+08:00 stderr F    /go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.8/pool.go:137 +0x34a
2022-11-29T05:10:46.51854021+08:00 stderr F
2022-11-29T05:10:46.518544755+08:00 stderr F goroutine 471 [select]:
2022-11-29T05:10:46.518564035+08:00 stderr F net/http/pprof.sleep(0xc007196600, 0xc0072f6000?)
2022-11-29T05:10:46.51857641+08:00 stderr F     /usr/local/go/src/net/http/pprof/pprof.go:98 +0x96
2022-11-29T05:10:46.518580516+08:00 stderr F net/http/pprof.Profile({0x40ee900, 0xc0072f6000}, 0x0?)
2022-11-29T05:10:46.518586644+08:00 stderr F    /usr/local/go/src/net/http/pprof/pprof.go:142 +0x325
2022-11-29T05:10:46.518590996+08:00 stderr F net/http.HandlerFunc.ServeHTTP(0x7f3cf4913650?, {0x40ee900?, 0xc0072f6000?}, 0x1362625?)
2022-11-29T05:10:46.518595631+08:00 stderr F    /usr/local/go/src/net/http/server.go:2084 +0x2f
2022-11-29T05:10:46.518599831+08:00 stderr F net/http.(*ServeMux).ServeHTTP(0xc0001ad399?, {0x40ee900, 0xc0072f6000}, 0xc007196600)
2022-11-29T05:10:46.518604821+08:00 stderr F    /usr/local/go/src/net/http/server.go:2462 +0x149
2022-11-29T05:10:46.518609119+08:00 stderr F net/http.serverHandler.ServeHTTP({0xc001c04b70?}, {0x40ee900, 0xc0072f6000}, 0xc007196600)
2022-11-29T05:10:46.518612972+08:00 stderr F    /usr/local/go/src/net/http/server.go:2916 +0x43b
2022-11-29T05:10:46.518616532+08:00 stderr F net/http.(*conn).serve(0xc0008483c0, {0x40f05b8, 0xc0007b3860})
2022-11-29T05:10:46.518620484+08:00 stderr F    /usr/local/go/src/net/http/server.go:1966 +0x5d7
2022-11-29T05:10:46.51862425+08:00 stderr F created by net/http.(*Server).Serve
2022-11-29T05:10:46.518636836+08:00 stderr F    /usr/local/go/src/net/http/server.go:3071 +0x4db
2022-11-29T05:10:46.518651974+08:00 stderr F
2022-11-29T05:10:46.518659682+08:00 stderr F goroutine 144 [select, 71 minutes]:
2022-11-29T05:10:46.518666363+08:00 stderr F google.golang.org/grpc.(*ccBalancerWrapper).watcher(0xc000fbf040)
2022-11-29T05:10:46.518673411+08:00 stderr F    /go/pkg/mod/google.golang.org/grpc@v1.46.0/balancer_conn_wrappers.go:112 +0x73
2022-11-29T05:10:46.518680222+08:00 stderr F created by google.golang.org/grpc.newCCBalancerWrapper
2022-11-29T05:10:46.518686484+08:00 stderr F    /go/pkg/mod/google.golang.org/grpc@v1.46.0/balancer_conn_wrappers.go:73 +0x22a
2022-11-29T05:10:46.518692499+08:00 stderr F
2022-11-29T05:10:46.518697412+08:00 stderr F goroutine 266 [chan receive]:
2022-11-29T05:10:46.518703206+08:00 stderr F github.com/panjf2000/ants/v2.(*Pool).purgePeriodically(0xc000482000)
2022-11-29T05:10:46.518709151+08:00 stderr F    /go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.8/pool.go:69 +0x8b
2022-11-29T05:10:46.51871347+08:00 stderr F created by github.com/panjf2000/ants/v2.NewPool
2022-11-29T05:10:46.518717696+08:00 stderr F    /go/pkg/mod/github.com/panjf2000/ants/v2@v2.4.8/pool.go:137 +0x34a
```

**Comment by liliu-z** [comment]
This is fixed in Knowhere 1.3.5.
